source/reference/failure-recovery.txt

```diff
@@ -0,0 +1,97 @@
+.. _c2c-failure-recovery:
+
+==========================
+Recovery From Sync Failure
+==========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+:program:`mongosync` temporarily alters some collection characteristics
+during synchronization. The original values are restored during the
+:ref:`commit process <c2c-api-commit>`. However, a network failure or
+some other problem could disrupt synchronization before the sync is
+committed. If the sync ends prematurely, you must manually update the
+temporary changes on the destination cluster.
+
+Manually revert the following temporary changes:
+
+- :ref:`Unique indexes set as non-unique <c2c-dr-unique>`
+- :ref:`TTL indexes with expireAfterSeconds set to MAX_INT
+  <c2c-dr-ttl>`
+- :ref:`Hidden indexes set as non-hidden <c2c-dr-hidden>`
+- :ref:`Capped collections with size/max set to the maximum possible
+  value <c2c-dr-capped>`
+
```

Comment:
Sorry, @alan-zheng-mdb pointed out that I missed write blocking in the google doc. So if write blocking is enabled as part of the /start, mongosync will set {setUserWriteBlockMode: 1, global: true} to block user writes against the destination cluster while the replication is ongoing. So if the replication is ended prematurely, users will have to manual unblock write on the destination before they can switch their application to the destination. To do that, they can do:
```
db.adminCommand(
    {
        setUserWriteBlockMode: 1,
        global: <boolean>
    }
)
```

We can link https://www.mongodb.com/docs/v6.0/reference/command/setUserWriteBlockMode/ for this

Reply:
Done